### ISSUE TITLE
Different behaviour when running code in the REPL or in Jupyter notebook

### ISSUE THREAD

**Issue by falbarelli** [tpapp/DynamicHMC.jl#86]
Hi, I also wrote a message on the [julia discourse board.](https://discourse.julialang.org/t/ann-dynamichmc-2-0/28314/10?u=falbarelli).
I'm on the latest julia version (1.2 stable) and I pulled `TransformVariables`, `LogDensityProblems` and `DynamicHMC` from the master branch. I'm using Julia.
I have reproduced the problem both on Ubuntu 16.04 and macOS 10.14.6.

It's probably a very stupid problem, but I can't figure it out!

## Self-contained example that demonstrates the problem

```julia
using LinearAlgebra
using StaticArrays
using TransformVariables
using LogDensityProblems
using DynamicHMC
using Parameters
using Random

function LogLikelihoodMin(ω,dyDep)
    
 dt = 0.1
 cD = [0 -1im ; 1im 0]
 H = (ω/2.) * cD
 rho=[0.05 0 ; 0 0.95]
 M = I - ((cD'*cD)/2) * dt + (cD * dyDep) - 1im * H * dt   
 newRho = M * rho * M'
 lklhood = real(tr(newRho))- (ω* dt/2)^2 
 return log(lklhood)

end

struct Experiment
      dyDep::Float64
end

function (problem::Experiment)((ω,)::NamedTuple{(:ω,)})
      @unpack dyDep  = problem        # extract the data
      LogLikelihoodMin(ω,dyDep)
end

dyDepObs=0.690691
p1 = Experiment(dyDepObs)

print(p1((ω=.4,)))

trans_single = as((ω=as(Real, 2, 4),))
P1 = TransformedLogDensity(trans_single, p1)
∇P1 = ADgradient(:ForwardDiff, P1)

results = mcmc_with_warmup(Random.GLOBAL_RNG, ∇P1, 1000)
```

## Output, expected outcome, comparison to other samplers

Basically, if I run this code in the REPL or launch it as a script it runs correctly, while if I copy and paste it in a Jupyter notebook I get the error

> ArgumentError: `nothing` should not be printed; use `show`, `repr`, or custom output instead.

See the whole stacktrace [here](https://pastebin.com/mQ9mRD9m).

**Comment by tpapp** [comment]
I could not replicate this.

Please make sure that DynamicHMC is at 2.0.0, eg with a 
```julia
import Pkg
Pkg.status()
```
cell in the notebook.

If this appears to be correct, please attach the Jupypter notebook as a file.

**Comment by falbarelli** [comment]
The version is `DynamicHMC v2.0.0 #master`.

Thanks again for the support and for your project. 
I found this behaviour very weird and I am afraid I am doing something stupid.
Find the the notebook [here](https://unimi2013-my.sharepoint.com/:u:/g/personal/francesco_albarelli_unimi_it/EaFbswfW83BNsRV9kjZILooBPai2ONVcMLKzG5rbuxzpQg?e=lhNrc8).

**Comment by falbarelli** [comment]
Oh I see that you have opened a [new issue](https://github.com/tpapp/DynamicHMC.jl/issues/87).
Yes, I think the problem is the output of the interactive progress report, if i add `reporter = NoProgressReport()` it runs fine.

**Comment by tpapp** [comment]
Yes, that's what I thought, but now I can confurm -- fixing #87 will provide a workaround for this.

The underlying issue in Julia appears to have been fixed in https://github.com/JuliaLang/julia/pull/32148, but I will just clean up the reporting in the meantime.

BTW,
```julia
@info "foo" nothing
```
is an MWE.

**Comment by falbarelli** [comment]
Oh thanks, now I understand, I didn't realize it was a more general issue in Julia.

**Comment by tpapp** [comment]
Thanks for reporting, I fixed it and will tag a release soon.
